### PR TITLE
Add style cop.json file

### PR DESCRIPTION
**Pull request by chidozieononiwu**

### CODE REVIEW

File: eng/stylecop.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
+    "settings": {
+        "documentationRules": {
+            "copyrightText": "Copyright (C) Microsoft Corporation. All rights reserved.\nLicensed under the MIT License",
+            "companyName": "Microsoft Corporation",
+            "xmlHeader": false,
+            "documentExposedElements": true,
```
Comment (pakrym) [inline eng/stylecop.json]: Why do we need all these documentExposed* parameters?

**Review by pakrym — APPROVED** [review]
Merge yours and I'll rebase.

File: eng/stylecop.json (hunk 1 above)
Comment (heaths) [reply eng/stylecop.json]: I'd keep the rules here minimal and rely more on .editorconfig. Unfortunately, .editorconfig won't fail the build, though, while this has the potential to if that's what we want (probably eventually).

**Review by heaths — APPROVED** [review]

File: eng/stylecop.json (hunk 1 above)
Comment (chidozieononiwu) [reply eng/stylecop.json]: Those serve to enforce XML documentations on Public members Interfaces and Internals types.

**Pull request merged**